api/envoy/api/v2/route/route.proto

```diff
@@ -615,7 +615,35 @@ message RouteAction {
     // 0.01% of requests to be mirrored. If the runtime key is specified in the
     // configuration but not present in runtime, 0 is the default and thus 0% of
     // requests will be mirrored.
-    string runtime_key = 2;
+    //
+    // .. attention::
+    //
+    //   **This field is deprecated**. Set the
+    //   :ref:`runtime_fraction
+    //   <envoy_api_field_route.RouteAction.RequestMirrorPolicy.runtime_fraction>` field instead.
+    string runtime_key = 2 [deprecated = true];
+
+    // If both :ref:`runtime_key
+    // <envoy_api_field_route.RouteAction.RequestMirrorPolicy.runtime_key>` and this field are not
+    // specified, all requests to the target cluster will be mirrored.
+    //
+    // If specified, this field takes precedence over the `runtime_key` field and requests must also
+    // fall under the percentage of matches indicated by this field.
+    //
+    // For some fraction N/D, a random number in the range [0,D) is selected. If the
+    // number is <= the value of the numberator N, or if the key is not present, the default
+    // value, the request will be mirrored.
+    //
+    // .. note::
+    //
+    //   Parsing this field is implemented such that the runtime key's data may be represented
+    //   as a :ref:`FractionalPercent <envoy_api_msg_type.FractionalPercent>` proto represented
+    //   as JSON/YAML and may also be represented as an integer with the assumption that the value
+    //   is an integral percentage out of 100. For instance, a runtime key lookup returning the
```

Comment:
It's probably not great tuning (and having to call out explicitly) JSON/YAML parsing behavior. We should have the APIs reflect a data model that makes sense and not tune for human input tweaks; there was always an API principle that the APIs are first and foremost intended to be for machine generation; fewer edge cases are better here. Also search for "machine generation" at https://github.com/envoyproxy/envoy/blob/master/api/STYLE.md.

Reply:
@htuch hmm can you provide some more guidance here? This was copy/pasted from another field in this file... are we sure we want to address changing the semantics/format of the `runtime_fraction` field(s) in this change?

Reply:
Oh, I see, this isn't about parsing the proto, but the ingest from the runtime. Please ignore, PEBKAC.

Reply:
Quick drive by comment: I think this description of how the runtime key is supposd to look is going to get copied to a lot of different places. Can we potentially move it into the runtime configuration and/or operation docs, and then just link there from each place? That will reduce copying?

Reply:
@mattklein123 yeah, good idea

Reply:
@rgs1 are you planning on doing that as part of this PR or a separate one?

Reply:
@htuch you are the boss, boss — though i'd be happier to do a follow-up PR to clean up the copy/pasta and go ahead and land this one as soon as I address your remaining comments...